### PR TITLE
Sprint 32 TLT-2014 Add a new command to dump emails sent through lti_emailer

### PR DESCRIPTION
**Pull request by rascalking**
* dumps a csv of all relevant details
* factored the event pull out to a library for use by other commands

Usage:
```
usage: manage.py mailgun_emails_sent [-h] [--version] [-v {0,1,2,3}]
                                     [--settings SETTINGS]
                                     [--pythonpath PYTHONPATH] [--traceback]
                                     [--no-color] --begin-time BEGIN_TIME
                                     --end-time END_TIME
                                     [--output-file OUTPUT_FILE]

Prints a list of emails sent in a given time window

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit
  -v {0,1,2,3}, --verbosity {0,1,2,3}
                        Verbosity level; 0=minimal output, 1=normal output,
                        2=verbose output, 3=very verbose output
  --settings SETTINGS   The Python path to a settings module, e.g.
                        "myproject.settings.main". If this isn't provided, the
                        DJANGO_SETTINGS_MODULE environment variable will be
                        used.
  --pythonpath PYTHONPATH
                        A directory to add to the Python path, e.g.
                        "/home/djangoprojects/myproject".
  --traceback           Raise on CommandError exceptions
  --no-color            Don't colorize the command output.
  --begin-time BEGIN_TIME
                        Beginning of the window to look for emails, as unix
                        timestamp
  --end-time END_TIME   End of the window to look for emails, as unix
                        timestamp
  --output-file OUTPUT_FILE
                        File to output the csv of emails sent to, defaults to
                        stdout
```

Sample output:
```
(lti_emailer)vagrant@precise64:~/tlt/lti_emailer(task/rascalking/tlt-2014/add-command-to-dump-email-details) $ python manage.py mailgun_emails_sent --begin 1443100620 --end 1442927820
timestamp,list,sender,subject,message-id
1443018234.343599,canvas-6760-2069@mg.dev.tlt.harvard.edu,david_bonner@harvard.edu,text inline test,BLUPR0701MB81737D7BBE335EDEB69AB6BE1440@BLUPR0701MB817.namprd07.prod.outlook.com
1443018236.402548,canvas-6760-2069@mg.dev.tlt.harvard.edu,david_bonner@harvard.edu,text inline test,BLUPR0701MB81737D7BBE335EDEB69AB6BE1440@BLUPR0701MB817.namprd07.prod.outlook.com
1443024438.403159,canvas-6760-2069@mg.dev.tlt.harvard.edu,david_bonner@harvard.edu,test test,BLUPR0701MB8176D5BF8F3656C09487CF8E1440@BLUPR0701MB817.namprd07.prod.outlook.com
1443024440.51515,canvas-6760-2069@mg.dev.tlt.harvard.edu,david_bonner@harvard.edu,test test,BLUPR0701MB8176D5BF8F3656C09487CF8E1440@BLUPR0701MB817.namprd07.prod.outlook.com
1443024634.650738,canvas-6760-2069@mg.dev.tlt.harvard.edu,david_bonner@harvard.edu,test test,SN2PR0701MB83059C1297F5DDFCF1F747CE1440@SN2PR0701MB830.namprd07.prod.outlook.com
1443024636.792531,canvas-6760-2069@mg.dev.tlt.harvard.edu,david_bonner@harvard.edu,test test,SN2PR0701MB83059C1297F5DDFCF1F747CE1440@SN2PR0701MB830.namprd07.prod.outlook.com
1443024701.079242,canvas-6760-2069@mg.dev.tlt.harvard.edu,david_bonner@harvard.edu,test test,SN2PR0701MB82983E1EA28D5D41144BB97E1440@SN2PR0701MB829.namprd07.prod.outlook.com
1443024703.135383,canvas-6760-2069@mg.dev.tlt.harvard.edu,david_bonner@harvard.edu,test test,SN2PR0701MB82983E1EA28D5D41144BB97E1440@SN2PR0701MB829.namprd07.prod.outlook.com
```

### CODE REVIEW

File: mailgun/management/commands/mailgun_emails_sent.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    def process_events(self, events, output_file):
+        rows = []
+        for event in events:
+            try:
+                sender = event['envelope']['sender']
+            except RuntimeError:
+                logger.exception('event {} missing a sender'.format(event))
+                continue
+
+            if sender.endswith(settings.LISTSERV_DOMAIN):
```
Comment (sapnamysore) [inline mailgun/management/commands/mailgun_emails_sent.py]: Can you clarify what kind of senders these are that are skipped

**Comment by eparker71** [comment]
:+1: Looks good

File: mailgun/management/commands/mailgun_emails_sent.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            try:       
+                list_address = None
+                for recipient in event['message']['recipients']:
+                    if (settings.LISTSERV_SECTION_ADDRESS_RE.match(recipient)
+                            or settings.LISTSERV_COURSE_ADDRESS_RE.match(recipient)):
+                        list_address = recipient
+                        break
+
+                row = [
+                    float(event['timestamp']),
```
Comment (sapnamysore) [inline mailgun/management/commands/mailgun_emails_sent.py]: It would be nice  to print readable datetime than timestamp?

**Comment by sapnamysore** [comment]
:+1:

File: mailgun/management/commands/mailgun_emails_sent.py (hunk 1 above)
Comment (rascalking) [inline mailgun/management/commands/mailgun_emails_sent.py]: The "accepted" events include the mails that we send out to all the members of the list.  That line checks for events that our mailer sends, and skips them.

Comment (sapnamysore) [inline mailgun/management/commands/mailgun_emails_sent.py]: got it. Thanks!

File: mailgun/management/commands/mailgun_emails_sent.py (hunk 2 above)
Comment (rascalking) [inline mailgun/management/commands/mailgun_emails_sent.py]: Sure.

**Pull request merged**